src/ngAria/aria.js

```diff
@@ -377,6 +377,25 @@ ngAriaModule.directive('ngShow', ['$aria', function($aria) {
             });
           }
         }
+
+        if (elem[0].nodeName === 'A') {
+          if ($aria.config('bindRoleForClick') && !attr.href && !attr.xlinkHref && !attr.role) {
+              elem.attr('role', 'link');
+          }
+          if ($aria.config('bindKeypress')) {
+            elem.on('keypress', function(event) {
+              var keyCode = event.which || event.keyCode;
+              var hasHref = (attr.href || attr.xlinkHref);
+              if ((keyCode === 32 || (keyCode === 13 && !hasHref)) && !attr.ngKeypress) {
```

Comment:
I don't think we should do anything (regardless the button clicked), if `hasHref` is truthy. What we are trying to do here is to "fix" accessibility on apps that (for whatever reason) are trying to use `<a>` elements as buttons (in which case they shouldn't provide an `href` value).

If someone applies both `href` and `ngClick`, I'm afraid they're on their own. Besides, different browsers handle the situation differently (e.g. IIRC Chrome first changes the URL and then executes the click handler and Firefox does the opposite (or it could be the other way around)).

`ngAria` should not be into that business. All we should be concerned with here is making `<a ng-click="...">...` behave like a "real" button wrt keyboard input.